### ISSUE TITLE
edgemesh agent not work because of Observed a panic: "invalid memory address or nil pointer dereference" 

### ISSUE THREAD

**Issue by wuminsure** [kubeedge/edgemesh#288]
**What happened**:

I install the edgemesh by helm,but the edgemesh agent is not wok,but the log not tell me what happend.the log is:

````
I0211 10:18:13.485505 1 server.go:62] Version: v0.0.0-master+$Format:%h$
I0211 10:18:13.485532 1 server.go:96] [1] New informers manager
I0211 10:18:13.486499 1 server.go:103] [2] Prepare agent to run
I0211 10:18:13.486671 1 netif.go:98] bridge device edgemesh0 already exists
I0211 10:18:13.487338 1 server.go:107] edgemesh-agent running on EdgeMode
I0211 10:18:13.487348 1 server.go:110] [3] Register beehive modules
I0211 10:18:13.487366 1 module.go:34] Module edgedns registered successfully
I0211 10:18:13.717821 1 module.go:34] Module edgeproxy registered successfully
W0211 10:18:13.717861 1 module.go:37] Module edgegateway is disabled, do not register
I0211 10:18:13.717943 1 manager.go:33] Use noCaManager to handle acl
I0211 10:18:13.717984 1 nocamanager.go:44] Private key exist, skip generate
I0211 10:18:13.724135 1 module.go:34] Module tunnelagent registered successfully
I0211 10:18:13.724177 1 server.go:119] [4] Install go-chassis plugins
I0211 10:18:13.724777 1 log.go:184] DEBUG: Installed strategy plugin: [Random].
I0211 10:18:13.724832 1 log.go:184] DEBUG: ConfigurationFactory Initiated
I0211 10:18:13.724866 1 log.go:184] INFO: Configuration files: []
I0211 10:18:13.724917 1 log.go:184] WARN: empty configurtion from [FileSource]
I0211 10:18:13.724936 1 log.go:184] INFO: invoke dynamic handler:FileSource
I0211 10:18:13.725012 1 log.go:184] INFO: archaius init success
I0211 10:18:13.725030 1 server.go:124] [5] Start informers manager
I0211 10:18:13.725267 1 log.go:184] INFO: create new watcher
W0211 10:18:13.726281 1 shared_informer.go:372] The sharedIndexInformer has started, run more than once is not allowed
W0211 10:18:13.726323 1 shared_informer.go:372] The sharedIndexInformer has started, run more than once is not allowed
W0211 10:18:13.726344 1 shared_informer.go:372] The sharedIndexInformer has started, run more than once is not allowed
W0211 10:18:13.726987 1 shared_informer.go:372] The sharedIndexInformer has started, run more than once is not allowed
I0211 10:18:14.326644 1 server.go:128] [6] Start all modules
I0211 10:18:14.326719 1 core.go:24] Starting module edgedns
I0211 10:18:14.326760 1 core.go:24] Starting module edgeproxy
I0211 10:18:14.326793 1 core.go:24] Starting module tunnelagent
I0211 10:18:14.326786 1 module.go:55] Runs CoreDNS v1.8.7 as a node-level dns
E0211 10:18:14.326821 1 runtime.go:78] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
goroutine 215 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic({0x1d2e060, 0x35e6330})
/code/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:74 +0x85
k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0x0})
/code/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:48 +0x75
panic({0x1d2e060, 0x35e6330})
/usr/local/go/src/runtime/panic.go:1038 +0x215
github.com/kubeedge/edgemesh/agent/pkg/chassis/controller.isConsistentHashLB(...)
/code/agent/pkg/chassis/controller/controller.go:137
github.com/kubeedge/edgemesh/agent/pkg/chassis/controller.(*ChassisController).drAdd(0x40d907, {0x1fd7d20, 0xc000ab21e0})
/code/agent/pkg/chassis/controller/controller.go:97 +0x4e
k8s.io/client-go/tools/cache.ResourceEventHandlerFuncs.OnAdd(...)
/code/vendor/k8s.io/client-go/tools/cache/controller.go:231
k8s.io/client-go/tools/cache.(*processorListener).run.func1()
/code/vendor/k8s.io/client-go/tools/cache/shared_informer.go:787 +0x9f
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0x7f6f836a0cf0)
/code/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:155 +0x67
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc000391f38, {0x23f09c0, 0xc00039a030}, 0x1, 0xc000584120)
/code/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:156 +0xb6
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0x0, 0x3b9aca00, 0x0, 0x37, 0xc000391fb0)
/code/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133 +0x89
k8s.io/apimachinery/pkg/util/wait.Until(...)
/code/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:90
k8s.io/client-go/tools/cache.(*processorListener).run(0xc000916200)
/code/vendor/k8s.io/client-go/tools/cache/shared_informer.go:781 +0x6b
k8s.io/apimachinery/pkg/util/wait.(*Group).Start.func1()
/code/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:73 +0x5a
created by k8s.io/apimachinery/pkg/util/wait.(*Group).Start
/code/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:71 +0x88
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x17087ce]
goroutine 215 [running]:
k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0x0})
/code/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:55 +0xd8
panic({0x1d2e060, 0x35e6330})
/usr/local/go/src/runtime/panic.go:1038 +0x215
github.com/kubeedge/edgemesh/agent/pkg/chassis/controller.isConsistentHashLB(...)
/code/agent/pkg/chassis/controller/controller.go:137
github.com/kubeedge/edgemesh/agent/pkg/chassis/controller.(*ChassisController).drAdd(0x40d907, {0x1fd7d20, 0xc000ab21e0})
/code/agent/pkg/chassis/controller/controller.go:97 +0x4e
k8s.io/client-go/tools/cache.ResourceEventHandlerFuncs.OnAdd(...)
/code/vendor/k8s.io/client-go/tools/cache/controller.go:231
k8s.io/client-go/tools/cache.(*processorListener).run.func1()
/code/vendor/k8s.io/client-go/tools/cache/shared_informer.go:787 +0x9f
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0x7f6f836a0cf0)
/code/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:155 +0x67
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc000391f38, {0x23f09c0, 0xc00039a030}, 0x1, 0xc000584120)
/code/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:156 +0xb6
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0x0, 0x3b9aca00, 0x0, 0x37, 0xc000391fb0)
/code/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133 +0x89
k8s.io/apimachinery/pkg/util/wait.Until(...)
/code/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:90
k8s.io/client-go/tools/cache.(*processorListener).run(0xc000916200)
/code/vendor/k8s.io/client-go/tools/cache/shared_informer.go:781 +0x6b
k8s.io/apimachinery/pkg/util/wait.(*Group).Start.func1()
/code/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:73 +0x5a
created by k8s.io/apimachinery/pkg/util/wait.(*Group).Start
/code/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:71 +0x88
````

**How to reproduce it (as minimally and precisely as possible)**:

I see the code of the error happend area,

````go
func isConsistentHashLB(dr *istioapi.DestinationRule) (ok bool) {
	switch dr.Spec.TrafficPolicy.LoadBalancer.LbPolicy.(type) {
	case *apiv1alpha3.LoadBalancerSettings_ConsistentHash:
		ok = true
	default:
		ok = false
	}
	return ok
}
````

it seem the *istioapi.DestinationRule not exist,I see the helm code of installing edgemesh ,it only install some crd of istio.

**Anything else we need to know?**:

I guess it is related to istio,but I do not know what shall I start with first? 

**Environment**:

- EdgeMesh version:1.9.0
- Kubernetes version (use `kubectl version`):1.22.5
- KubeEdge version(e.g. `cloudcore --version` and `edgecore --version`):
- <details><summary>Cloud nodes Environment:</summary>

    - Hardware configuration (e.g. `lscpu`):intel
    - OS (e.g. `cat /etc/os-release`):centos8
    - Kernel (e.g. `uname -a`):
    - Go version (e.g. `go version`): image from docker hub
    - Others:

  </details>
- <details><summary>Edge nodes Environment:</summary>

    - edgecore version (e.g. `edgecore --version`):1.8.2
    - Hardware configuration (e.g. `lscpu`):intel
    - OS (e.g. `cat /etc/os-release`):centos8
    - Kernel (e.g. `uname -a`):
    - Go version (e.g. `go version`): image from docker hub
    - Others:

  </details>

**Comment by Poorunga** [comment]
Can you try: kubectl get crds, then show me logs?

**Comment by wuminsure** [comment]
> Can you try: kubectl get crds, then show me logs?
````bash
(base) [root@k8smaster ~]# kubectl get crds 
NAME                                                 CREATED AT
adapters.config.istio.io                             2021-11-18T01:11:17Z
apikeys.config.istio.io                              2021-11-18T01:11:17Z
applications.app.k8s.io                              2021-11-18T01:11:46Z
attributemanifests.config.istio.io                   2021-11-18T01:11:17Z
authorizations.config.istio.io                       2021-11-18T01:11:17Z
bypasses.config.istio.io                             2021-11-18T01:11:17Z
certificaterequests.cert-manager.io                  2021-11-18T01:11:54Z
certificates.cert-manager.io                         2021-11-18T01:11:55Z
certificates.certmanager.k8s.io                      2021-11-18T01:11:17Z
certificates.networking.internal.knative.dev         2021-11-18T02:33:47Z
challenges.acme.cert-manager.io                      2021-11-18T01:11:55Z
challenges.certmanager.k8s.io                        2021-11-18T01:11:17Z
checknothings.config.istio.io                        2021-11-18T01:11:17Z
circonuses.config.istio.io                           2021-11-18T01:11:18Z
cloudwatches.config.istio.io                         2021-11-18T01:11:18Z
clusterissuers.cert-manager.io                       2021-11-18T01:11:55Z
clusterissuers.certmanager.k8s.io                    2021-11-18T01:11:18Z
clusterobjectsyncs.reliablesyncs.kubeedge.io         2021-11-19T01:00:50Z
clusterrbacconfigs.rbac.istio.io                     2021-11-18T01:11:18Z
compositecontrollers.metacontroller.k8s.io           2021-11-18T02:33:39Z
configurations.serving.knative.dev                   2021-11-18T02:33:47Z
controllerrevisions.metacontroller.k8s.io            2021-11-18T02:33:39Z
decoratorcontrollers.metacontroller.k8s.io           2021-11-18T02:33:39Z
deniers.config.istio.io                              2021-11-18T01:11:18Z
destinationrules.networking.istio.io                 2021-11-18T01:11:18Z
devicemodels.devices.kubeedge.io                     2021-11-19T01:00:50Z
devices.devices.kubeedge.io                          2021-11-19T01:00:50Z
dogstatsds.config.istio.io                           2021-11-18T01:11:18Z
edges.config.istio.io                                2021-11-18T01:11:18Z
envoyfilters.networking.istio.io                     2021-11-18T01:11:19Z
experiments.kubeflow.org                             2021-11-18T02:34:08Z
fluentds.config.istio.io                             2021-11-18T01:11:19Z
gateways.networking.istio.io                         2021-11-18T01:11:19Z
handlers.config.istio.io                             2021-11-18T01:11:19Z
httpapispecbindings.config.istio.io                  2021-11-18T01:11:19Z
httpapispecs.config.istio.io                         2021-11-18T01:11:19Z
images.caching.internal.knative.dev                  2021-11-18T02:33:47Z
inferenceservices.serving.kubeflow.org               2021-11-18T02:34:04Z
ingresses.networking.internal.knative.dev            2021-11-18T02:33:47Z
instances.config.istio.io                            2021-11-18T01:11:19Z
issuers.cert-manager.io                              2021-11-18T01:11:55Z
issuers.certmanager.k8s.io                           2021-11-18T01:11:19Z
kubernetesenvs.config.istio.io                       2021-11-18T01:11:20Z
kuberneteses.config.istio.io                         2021-11-18T01:11:20Z
listcheckers.config.istio.io                         2021-11-18T01:11:20Z
listentries.config.istio.io                          2021-11-18T01:11:20Z
logentries.config.istio.io                           2021-11-18T01:11:20Z
memquotas.config.istio.io                            2021-11-18T01:11:20Z
meshpolicies.authentication.istio.io                 2021-11-18T01:11:20Z
metrics.autoscaling.internal.knative.dev             2021-11-18T02:33:48Z
metrics.config.istio.io                              2021-11-18T01:11:20Z
noops.config.istio.io                                2021-11-18T01:11:20Z
notebooks.kubeflow.org                               2021-11-18T02:33:46Z
objectsyncs.reliablesyncs.kubeedge.io                2021-11-19T01:00:50Z
opas.config.istio.io                                 2021-11-18T01:11:20Z
orders.acme.cert-manager.io                          2021-11-18T01:11:55Z
orders.certmanager.k8s.io                            2021-11-18T01:11:21Z
podautoscalers.autoscaling.internal.knative.dev      2021-11-18T02:33:48Z
poddefaults.kubeflow.org                             2021-11-18T02:33:42Z
policies.authentication.istio.io                     2021-11-18T01:11:21Z
profiles.kubeflow.org                                2021-11-18T02:34:22Z
prometheuses.config.istio.io                         2021-11-18T01:11:21Z
pytorchjobs.kubeflow.org                             2021-11-18T02:33:46Z
quotas.config.istio.io                               2021-11-18T01:11:21Z
quotaspecbindings.config.istio.io                    2021-11-18T01:11:21Z
quotaspecs.config.istio.io                           2021-11-18T01:11:21Z
rbacconfigs.rbac.istio.io                            2021-11-18T01:11:21Z
rbacs.config.istio.io                                2021-11-18T01:11:21Z
redisquotas.config.istio.io                          2021-11-18T01:11:21Z
reportnothings.config.istio.io                       2021-11-18T01:11:22Z
revisions.serving.knative.dev                        2021-11-18T02:33:48Z
routes.serving.knative.dev                           2021-11-18T02:33:48Z
ruleendpoints.rules.kubeedge.io                      2021-11-19T01:00:50Z
rules.config.istio.io                                2021-11-18T01:11:22Z
rules.rules.kubeedge.io                              2021-11-19T01:00:50Z
scheduledsparkapplications.sparkoperator.k8s.io      2021-11-23T01:04:46Z
scheduledworkflows.kubeflow.org                      2021-11-18T02:34:20Z
seldondeployments.machinelearning.seldon.io          2021-11-18T02:34:23Z
serverlessservices.networking.internal.knative.dev   2021-11-18T02:33:48Z
serviceentries.networking.istio.io                   2021-11-18T01:11:22Z
servicerolebindings.rbac.istio.io                    2021-11-18T01:11:22Z
serviceroles.rbac.istio.io                           2021-11-18T01:11:22Z
services.serving.knative.dev                         2021-11-18T02:33:48Z
sidecars.networking.istio.io                         2021-11-18T01:11:22Z
signalfxs.config.istio.io                            2021-11-18T01:11:22Z
solarwindses.config.istio.io                         2021-11-18T01:11:22Z
sparkapplications.sparkoperator.k8s.io               2021-11-23T01:04:46Z
stackdrivers.config.istio.io                         2021-11-18T01:11:23Z
statsds.config.istio.io                              2021-11-18T01:11:23Z
stdios.config.istio.io                               2021-11-18T01:11:23Z
suggestions.kubeflow.org                             2021-11-18T02:34:08Z
templates.config.istio.io                            2021-11-18T01:11:23Z
tfjobs.kubeflow.org                                  2021-11-18T02:34:06Z
tracespans.config.istio.io                           2021-11-18T01:11:23Z
trials.kubeflow.org                                  2021-11-18T02:34:08Z
viewers.kubeflow.org                                 2021-11-18T02:34:18Z
virtualservices.networking.istio.io                  2021-11-18T01:11:23Z
workflows.argoproj.io                                2021-11-18T02:33:40Z
zipkins.config.istio.io                              2021-11-18T01:11:23Z
````

**Comment by Poorunga** [comment]
Has your cluster already created some DestinationRule resources? Use `kubectl get dr --all-namespaces` to get them. BTW, this nil pointer dereference problem is indeed an issue.

**Comment by Poorunga** [comment]
OK, I will to fix this bug.

**Comment by Poorunga** [comment]
fix in PR #295 

/close
